Code/Tools/AssetProcessor/native/ui/BuilderData.cpp

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#include <native/ui/BuilderData.h>
+#include <utilities/AssetUtilEBusHelper.h>
+#include <native/ui/BuilderDataItem.h>
+#include <AzToolsFramework/AssetDatabase/AssetDatabaseConnection.h>
+#include <AzCore/StringFunc/StringFunc.h>
+#include <AzCore/std/smart_ptr/weak_ptr.h>
+
+namespace AssetProcessor
+{
+    const char builderNotFoundWarningMessage[] =
+        "Found a %s metric entry with builder %s \"%s\", but Asset Processor does not recognize this "
+        "builder. Ensure this builder is in the asset folders and its name is shown "
+        "in the Builders tab. If this builder was removed intentionally in the past, you can safely ignore this "
+        "warning.\n";
+
+    BuilderData::BuilderData(AZStd::shared_ptr<AzToolsFramework::AssetDatabase::AssetDatabaseConnection> dbConnection, QObject* parent)
+        : m_dbConnection(dbConnection)
+        , QObject(parent)
+    {
+    }
+
+    void BuilderData::Reset()
+    {
+        BuilderInfoList builders;
+        AssetBuilderInfoBus::Broadcast(&AssetBuilderInfoBus::Events::GetAllBuildersInfo, builders);
+
+        m_root.reset(new BuilderDataItem(BuilderDataItem::ItemType::InvisibleRoot, "", 0, 0, AZStd::weak_ptr<BuilderDataItem>()));
+
+        m_allBuildersMetrics.reset(new BuilderDataItem(BuilderDataItem::ItemType::Builder, "All builders", 0, 0, m_root));
+        m_allBuildersMetrics->InitializeBuilder(m_allBuildersMetrics);
+        m_singleBuilderMetrics.clear();
+        m_builderGuidToIndex.clear();
+        m_builderNameToIndex.clear();
+        m_currentSelectedBuilderIndex = aznumeric_cast<int>(BuilderSelection::AllBuilders);
+
+        for (int i = 0; i < builders.size(); ++i)
+        {
+            auto builder = m_singleBuilderMetrics.emplace_back(
+                new BuilderDataItem(BuilderDataItem::ItemType::Builder, builders[i].m_name, 0, 0, m_root));
+            builder->InitializeBuilder(builder);
+            m_builderGuidToIndex[builders[i].m_busId] = i;
+            m_builderNameToIndex[builders[i].m_name] = i;
+        }
+
+        // CreateJobs stat
+        m_dbConnection->QueryStatLikeStatName(
+            "CreateJobs,%",
+            [this](AzToolsFramework::AssetDatabase::StatDatabaseEntry entry)
+            {
+                AZStd::vector<AZStd::string> tokens;
+                AZ::StringFunc::Tokenize(entry.m_statName, tokens, ',');
```

Comment:
strange to me that you need to process the string like this  coming from an internal api?

Reply:
The metric measurement code (StatCapture) is intentionally made to be loosely coupled with the task being measured (e.g. CreateJobs and ProcessJob), so that if the measuring needs to be changed or is no longer needed, disabling/removing it is easy. An entry only has a StatName, a StatValue (the measurement), and a timestamp, and the only place to relate the task and the measurement is through the StatName.

The downside of this is what's happening here: after retrieving stat entries from the database, it still has to parse the stat name and link the stat with the corresponding task.